### ISSUE TITLE
💅 BIOME_CONFIG_PATH does not work

### ISSUE THREAD

**Issue by coolemur** [biomejs/biome#6160]
### Environment information

```bash
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.18.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0
```

### Rule name

config

### Playground link

?

### Expected result

Does not work:
`BIOME_CONFIG_PATH=/Users/Mantas/.config/biome.json biome lint test.ts`

Works:
`biome lint test.ts --config-path=/Users/Mantas/.config/biome.json`

Expected: config to be picked up when BIOME_CONFIG_PATH env var is set

https://biomejs.dev/guides/configure-biome/ documentation says it should work:
"Biome commands support using the --config-path option and the BIOME_CONFIG_PATH environment variable."

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by coolemur** [comment]
https://github.com/coolemur/biome-repro-1748546888955

**Comment by ematipico** [comment]
What do you this when you run `BIOME_CONFI_PATH=/path biome rage`? Does Biome print something, or is there an error?
